lib/yerror.js

```diff
@@ -0,0 +1,11 @@
+const util = require('util')
+
+function YError (msg) {
+  this.name = 'YError'
+  this.message = msg || 'yargs error'
+  this.stack = (new Error()).stack
```

Comment:
I think this last line would be better as:

```
  Error.captureStackTrace(this, YError)
```
By using `captureStackTrace` you won't get the `YError` constructor in the call stack.